mongodb-driver/src/test/java/org/eclipse/jnosql/communication/mongodb/document/MongoDBDocumentCollectionManagerTest.java

```diff
@@ -513,6 +520,390 @@ public void shouldSaveMap() {
         Assertions.assertNotNull(map);
     }
 
+    @Test
+    public void shouldSelect() {
+        entityManager.insert(getEntity());
+        CriteriaQuery<Person> criteriaQuery = new DefaultCriteriaQuery(Person.class);
+        SelectQuery<Person> select = criteriaQuery.select();
+        assertTrue(entityManager.executeQuery(select).toArray().length == 1);
+    }
+
+    @Test
+    public void shouldSelectByAttribute() {
+        entityManager.insert(getEntity());
+        CriteriaQuery<Person> criteriaQuery = new DefaultCriteriaQuery(Person.class);
```

Comment:
the code, in general, looks good, but, the communication layer should not have access to an entity class such as ```Person```

Reply:
I believe if we only move it as either a mapping class or include it as an extension:

https://github.com/eclipse/jnosql-mapping-extension/tree/master/mongodb-extension